client/dashboard/domains/index.tsx

```diff
@@ -1,41 +1,48 @@
 import { DomainSubtype } from '@automattic/api-core';
 import { domainsQuery } from '@automattic/api-queries';
 import { useQuery } from '@tanstack/react-query';
-import { DataViews, filterSortAndPaginate } from '@wordpress/dataviews';
+import { filterSortAndPaginate } from '@wordpress/dataviews';
 import { __ } from '@wordpress/i18n';
-import { useState } from 'react';
 import { useAuth } from '../app/auth';
 import { useAppContext } from '../app/context';
+import { usePersistentView, DataViews } from '../app/dataviews';
+import { domainsRoute } from '../app/router/domains';
 import { DataViewsCard } from '../components/dataviews-card';
 import { OptInWelcome } from '../components/opt-in-welcome';
 import { PageHeader } from '../components/page-header';
 import PageLayout from '../components/page-layout';
 import { AddDomainButton } from './add-domain-button';
 import { useActions, useFields, DEFAULT_VIEW, DEFAULT_LAYOUTS } from './dataviews';
-import type { DomainsView } from './dataviews';
 import type { DomainSummary } from '@automattic/api-core';
 
 export function getDomainId( domain: DomainSummary ): string {
 	return `${ domain.domain }-${ domain.blog_id }`;
 }
 
+const defaultView = {
+	...DEFAULT_VIEW,
+	filters: [
+		{
+			field: 'owner',
+			operator: 'isAny' as const,
+			value: [ 'owned-by-me' ],
+		},
+	],
+};
+
 function Domains() {
 	const { user } = useAuth();
 	const { queries } = useAppContext();
 	const fields = useFields();
 	const { data: sites } = useQuery( queries.sitesQuery() );
 	const actions = useActions( { user, sites } );
-	const [ view, setView ] = useState< DomainsView >( () => ( {
-		...DEFAULT_VIEW,
-		type: 'table',
-		filters: [
-			{
-				field: 'owner',
-				operator: 'isAny',
-				value: [ 'owned-by-me' ],
-			},
-		],
-	} ) );
+	const searchParams = domainsRoute.useSearch();
+
+	const { view, updateView, resetView } = usePersistentView( {
```

Comment:
Could have been useView like `@wordpress`, any idea for the different name?

Reply:
Oops. It was `useView()` before. But while working on integrating this package to existing codebase, I sometimes got confused myself because the existing code already uses hooks like `useFields()`, `useActions()`, and `useView()` itself... so I improvised and renamed it to `usePersistentView()` so that it's clear what's different 😛 but I forgot that `@wordpress/views` uses `useView()` 🤦 

I can revert it back to `useView()` if this is a strong argument to make it to have the same as Core?